### PR TITLE
Fix true spelling errors across all source and header files within the SDK and add exceptions for false positives.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Part of https://github.com/Azure/azure-sdk-for-cpp/issues/1277, checking what types of warnings the CI emits.

Verified all SDK product `.cpp`, `.hpp`, `.txt`, and `.md` files, primarily focused on azure-core. They are all clean now. There are some exceptions that needs to be added for keyvault and storage, but they are false positives, so not a concern.

> `cspell lint --config .vscode/cspell.json *.hpp */**/*.hpp`

CSpell: Files checked: 188, Issues found: 0 in 0 files

> `cspell lint --config .vscode/cspell.json *.cpp */**/*.cpp`

CSpell: Files checked: 186, Issues found: 88 in 15 files (keyvault and storage false positives, or tests)

> `cspell lint --config .vscode/cspell.json *.md */**/*.md`

CSpell: Files checked: 45, Issues found: 5 in 2 files (eng/common)

> `cspell lint --config .vscode/cspell.json *.txt */**/*.txt`

CSpell: Files checked: 44, Issues found: 0 in 0 files

> `cspell lint --config .vscode/cspell.json *.* */**/*`

CSpell: Files checked: 646, Issues found: 328 in 69 files (most of these are in eng\docs\api\assets\style.css or eng/common)

Deprioritize and ignored the errors from the test files (including test resource json files), and `eng/common` since those need to be centrally fixed.

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Comment by ahsonkhan** [comment]
@danieljurek the spell checking CI leg doesn't work when we have a lot of files changed. It fails with `The command line is too long.`:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=879540&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=ccad100f-2904-5913-2083-c525ad929a03

```text
Starting: Check spelling (cspell)
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.180.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
Formatted command: . 'D:\a\_work\1\s\eng\common\scripts\check-spelling-in-changed-files.ps1' -TargetBranch "origin/$("master" -replace "refs/heads/")" -SourceBranch HEAD -CspellConfigPath ./.vscode/cspell.json
========================== Starting Command Output ===========================
"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_work\_temp\13887517-c5ce-4eab-b69e-ab169ca42783.ps1'"
git diff --diff-filter=d --name-only origin/master HEAD
Git Detected 364 changed file(s). Files checked by cspell may exclude files according to cspell.json
npx cspell --config ./.vscode/cspell.json D:\a\_work\1\s\samples\integration\cmake-fetch-content\src\main.cpp D:\a\_work\1\s\sdk\core\azure-core\inc\azure\core.hpp D:\a\_work\1\s\sdk\core\azure-core\inc\azure\core\base64.hpp D:\a\_work\1\s\sdk\core\azure-core\inc\azure\core\case_insensitive_containers.hpp 
...
D:\a\_work\1\s\sdk\template\azure-template\inc\azure\template\template_client.hpp D:\a\_work\1\s\sdk\template\azure-template\src\private\package_version.hpp D:\a\_work\1\s\sdk\template\azure-template\src\template_client.cpp D:\a\_work\1\s\sdk\template\azure-template\test\ut\macro_guard.cpp D:\a\_work\1\s\sdk\template\azure-template\test\ut\template_test.cpp
The command line is too long.
Finishing: Check spelling (cspell)
```

**Comment by ahsonkhan** [comment]
Verified all SDK product `.cpp`, `.hpp`, `.txt`, and `.md` files, primarily focused on azure-core. They are all clean now. There are some exceptions that needs to be added for keyvault and storage, but they are false positives, so not a concern.

> cspell lint --config .vscode/cspell.json *.hpp */**/*.hpp

CSpell: Files checked: 188, Issues found: 0 in 0 files

> cspell lint --config .vscode/cspell.json *.cpp */**/*.cpp

CSpell: Files checked: 186, Issues found: 88 in 15 files (keyvault and storage false positives, or tests)

> cspell lint --config .vscode/cspell.json *.md */**/*.md

CSpell: Files checked: 45, Issues found: 5 in 2 files (eng/common)

> cspell lint --config .vscode/cspell.json *.txt */**/*.txt

CSpell: Files checked: 44, Issues found: 0 in 0 files

> cspell lint --config .vscode/cspell.json *.* */**/*

CSpell: Files checked: 646, Issues found: 328 in 69 files (most of these are in eng\docs\api\assets\style.css or eng/common)

File: eng/CHECKENFORCER [hunk 1]
```diff
@@ -28,7 +28,7 @@ message: >
   /check-enforcer evaluate
   ```
   
-  Typically evaulation only takes a few seconds. If you know that your pull
+  Typically evaluation only takes a few seconds. If you know that your pull
```
Comment (ahsonkhan) [inline eng/CHECKENFORCER:31]: @mitchdenny is it OK to modify this check enforcer file like this? Do other places/repos need to be fixed?

File: eng/scripts/Get-PkgVersion.ps1 [hunk 2]
```diff
@@ -11,7 +11,7 @@ $repoRoot = Resolve-Path "$PSScriptRoot/../..";
 . (Join-Path ${repoRoot} eng common scripts logging.ps1)
 . (Join-Path ${repoRoot} eng scripts SdkVersion-Common.ps1)
 
-$versionFileLocation = Get-VersionHppLocaiton `
+$versionFileLocation = Get-VersionHppLocation `
```
Comment (ahsonkhan) [inline eng/scripts/Get-PkgVersion.ps1:14]: At first, I though, how could the package version scripts be working with this typo. Turns out it was consistently incorrect (including the function definition), so at least that was good :)

**Review by Jinming-Hu — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]
@ahsonkhan Should this be auto-merged?

**Comment by ahsonkhan** [comment]
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1277

**Pull request merged**